src/cavendish_particle_tracks/_decay_angles_dialog.py

```diff
@@ -80,9 +80,27 @@ def __init__(self, parent=None):
     def _setup_decayangles_layer(self):
         """Create a shapes layer and add three lines to measure the Lambda, p and pi tracks"""
 
-        lambda_line = np.array([[100, 100], [200, 300]])
-        proton_line = np.array([[200, 300], [300, 400]])
-        pion_line = np.array([[200, 300], [210, 400]])
+        origin_x = self.parent.viewer.camera.center[1]
+        origin_y = self.parent.viewer.camera.center[2]
+
+        lambda_line = np.array(
+            [
+                [origin_x + -100, origin_y + -100],
+                [origin_x + 100, origin_y + 200],
+            ]
+        )
+        proton_line = np.array(
+            [
+                [origin_x + 100, origin_y + 200],
+                [origin_x + 200, origin_y + 300],
+            ]
+        )
+        pion_line = np.array(
+            [
+                [origin_x + 100, origin_y + 200],
+                [origin_x + 110, origin_y + 300],
+            ]
+        )
```

Comment:
Is it possible to add labels to the lines?

Reply:
Yep. Would fix the issue of the lines being invisible when zoomed out.